server/functions/timestamp.go

```diff
@@ -141,3 +142,39 @@ var timestamp_cmp = framework.Function2{
 		return int32(ab.Compare(bb)), nil
 	},
 }
+
+// FormatTimestampWithBC formats a time.Time that may represent BC dates (negative years)
+// PostgreSQL represents BC years as negative years in time.Time, but Go's Format() doesn't handle this correctly
+// tz is optional timezone value to be appended to formatted value
+func FormatTimestampWithBC(t time.Time, hasTZ bool) string {
```

Comment:
We also have `formatDateWithBC`, which seems very similar. The `layout` is different and `positiveTime` uses zeroes, but it seems like we could just have a boolean parameter that determines which `layout` we use, and also which `positiveTime` variant we use, since everything else is the same.